homeassistant/components/philips_js/light.py

```diff
@@ -0,0 +1,174 @@
+
+    def __init__(self, coordinator: PhilipsTVDataUpdateCoordinator):
+        """Initialize light."""
+        self._tv = coordinator.api
+        self._hsv = None
+        self._system = coordinator.system
+        self._coordinator = coordinator
+        super().__init__(coordinator)
+
+        self._update_from_coordinator()
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._system["name"]
+
+    @property
+    def unique_id(self):
+        """Return unique identifier if known."""
+        return self._system["serialnumber"]
+
+    @property
+    def supported_features(self):
+        """Return supported features on this light."""
+        return SUPPORT_EFFECT | SUPPORT_COLOR | SUPPORT_BRIGHTNESS
+
+    @property
+    def should_poll(self):
+        """Device should be polled."""
+        return True
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects."""
+        return ["internal", "manual", "expert"]
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._coordinator.ambilight_mode
+
+    @property
+    def hs_color(self):
+        """Return the hue and saturation color value [float, float]."""
+        if self._hsv:
+            return self._hsv[:2]
+
+    @property
+    def brightness(self):
+        """Return the hue and saturation color value [float, float]."""
+        if self._hsv:
+            return self._hsv[2] * 255 / 100
+
+    @property
+    def is_on(self):
+        """Return if the light is turned on."""
+        return self._tv.on
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return {
+            "name": self._system["name"],
+            "identifiers": {
+                (DOMAIN, self._system["serialnumber"]),
+            },
+            "model": self._system["model"],
+            "manufacturer": "Philips",
+            "sw_version": self._system["softwareversion"],
+        }
+
+    def _update_from_coordinator(self):
+        if self._coordinator.ambilight_mode == "internal":
+            data = self._coordinator.ambilight_processed
+        else:
+            data = self._coordinator.ambilight_cached
+
+        if data:
+            color_r, color_g, color_b = _average_pixels(data)
+            self._hsv = color_RGB_to_hsv(color_r, color_g, color_b)
+        else:
+            self._hsv = None
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_from_coordinator()
+        super()._handle_coordinator_update()
+
+    def turn_on(self, **kwargs) -> None:
+        """Turn the bulb on."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS)
+        hs_color = kwargs.get(ATTR_HS_COLOR)
+        effect = kwargs.get(ATTR_EFFECT)
+
+        if not self._tv.on:
+            raise Exception("Light is not turned on")
```

Comment:
This is a bit unusual. Would it be appropriate to indicate more clearly why this doesn't work and what should be done instead? I'm guessing its something like "Amblight can only be set if the TV is on."?

Reply:
Well yes. The TV can not be turned on via the API. We provide an event on the media player to hook up an ir blaster  to turn on. But as the light is currently set up. It's always on if the TV is turned on. It's the EFFECT that actually controls how the light behaves.

Reply:
I've reworded it.